starfish/image/_stack.py

```diff
@@ -335,29 +419,38 @@ def apply(self, func, is_volume=False, **kwargs):
             numpy.ndarray. If inplace is True, must return an array of the same shape.
         is_volume : bool
             (default False) If True, pass 3d volumes (x, y, z) to func
-        inplace : bool
+        in_place : bool
             (default True) If True, function is executed in place. If n_proc is not 1, the tile or
-            volume will be copied once during execution. Not currently implemented.
+            volume will be copied once during execution. If false, the outputs of the function executed on individual
+            tiles or volumes will be output as a list
+        verbose : bool
+            If True, report on the percentage completed (default = False) during processing
         kwargs : dict
             Additional arguments to pass to func
 
         Returns
         -------
-        Optional[ImageStack]
-            If inplace is False, return a new ImageStack containing the output of apply
+        Optional[List]
+            If inplace is False, return the results of applying func to stored image data
         """
-        mapfunc = map  # TODO: ambrosejcarr posix-compliant multiprocessing
+        mapfunc: Callable = map  # TODO: ambrosejcarr posix-compliant multiprocessing
         indices = list(self._iter_indices(is_volume=is_volume))
-        tiles = self._iter_tiles(indices)
 
-        applyfunc = partial(func, **kwargs)
+        if verbose:
+            tiles = tqdm(self._iter_tiles(indices))
+        else:
+            tiles = self._iter_tiles(indices)
 
+        applyfunc: Callable = partial(func, **kwargs)
         results = mapfunc(applyfunc, tiles)
 
+        if not in_place:
+            return list(results)
```

Comment:
is this a useful result since you lose the indices of the tiles?

Reply:
Good catch, will fix.